Templates/Common-Backend-Scripts/utilities/fetchBuildDependencies.groovy

```diff
@@ -0,0 +1,339 @@
+			def String artifactRelPath
+			if (dependency.version.startsWith("rel-")){
+				artifactRelPath="${repositoryName}/main/release/${dependency.version}"
+				artifactReference="${artifactRelPath}/${dependency.name}.tar"
+				artifactUrl="${props.artifactRepositoryUrl}/${artifactReference}"
+			} else {
+				artifactRelPath="${repositoryName}"
+				artifactReference="${artifactRelPath}/${dependency.name}.tar"
+				artifactUrl="${props.artifactRepositoryUrl}/${artifactReference}"
+			}
+			println("*** Fetching package '${dependency.name}:${dependency.version}' ")
+
+			// Generating information for documentation in yaml file of retrieved dependencies
+			if (dependency.name != applicationDescriptor.application) {
+				
+				ExternalDependency externalDependency = new ExternalDependency()
+				// Map information to External Dependency Record
+				externalDependency.name = dependency.name // dependency name
+				externalDependency.type = dependency.type // dependency type
+				externalDependency.properties = new HashSet()
+				
+				Property p_uri = new Property()
+				p_uri.key = "uri"
+				p_uri.value = artifactReference
+				externalDependency.properties.add(p_uri)
+				Property p_version = new Property()
+				p_version.key = "version"
+				p_version.value = dependency.version
+				externalDependency.properties.add(p_version)
+				
+				// Store external dependency information
+				externalDependencies.add(externalDependency)
+			}
+
+			String tarFile = "${cacheFolder}/${artifactReference}"
+			String includeFolder = "${importFolder}/${dependency.name}"
+
+			if (new File(tarFile).exists()) {
+				println("** Package was already found in package cache at '${cacheFolder}/${artifactRelPath}'")
+			} else {
+				String user = props.artifactRepositoryUser
+				String password = props.artifactRepositoryPassword
+
+				if (!(new File("${cacheFolder}/${artifactRelPath}").exists())) (new File("${cacheFolder}/${artifactRelPath}")).mkdirs()
+
+				println("** Downloading application package '$artifactUrl' from Artifact Repository into ${cacheFolder}/${artifactRelPath}.")
+				def rc = artifactRepositoryHelpers.download(artifactUrl, tarFile as String, user, password, true)
+				println "download complete $rc" // TODO: Error handling in helper
+			}
+
+
+			File includeFolderFile = new File(includeFolder)
+			if (!includeFolderFile.exists()) {
+				includeFolderFile.mkdirs()
+			}
+
+
+			println("** Expanding tar file '$tarFile' to '$includeFolder' ")
+
+			def processCmd = [
+				"/bin/sh",
+				"-c",
+				"tar -C $includeFolder -xvf $tarFile"
+			]
+
+			def rc = runProcess(processCmd)
+			if (rc != 0) {
+				println("** [ERROR] Failed to untar '$tarFile' to '$includeFolder' with rc=$rc")
+				System.exit(1)
+			}
+		} else {
+			println("* Dependency Types other than 'artifactrepository' are not yet implemented. Exiting.")
+			System.exit(1)
+		}
+	}
+}
+
+// Do we actually need this or can this just be obtained from the dependencies?
```

Comment:
The Baseline section of the AD file is meant to describe "dependency" on itself for a given application. It can serve two purposes: the first one is downloading the packages from earlier version that can be used as a base for artifacts that are not rebuilt (i.e., private object decks, generated include files from BMS, ...). The second one is to detect the baselineReference tag during an impact build.

I think the next section is necessary. Maybe it can be optimized to be turned into a method, to minimize the common code with the previous section.